echopype/utils/coding.py

```diff
@@ -142,16 +144,19 @@ def set_zarr_encodings(ds: xr.Dataset, compression_settings: dict) -> dict:
     # create zarr specific encoding
     encoding = dict()
     for name, val in ds.variables.items():
-        val_encoding = val.encoding
-        val_encoding.update(get_zarr_compression(val, compression_settings))
+        encoding[name] = {**val.encoding}
+        encoding[name].update(get_zarr_compression(val, compression_settings))
 
-        # If data array is not a dask array yet,
-        # create a custom chunking encoding
-        # currently defaults to 100MB
-        if not ds.chunks and len(val.shape) > 0:
+        # Always optimize chunk if not specified already
+        # user can specify desired chunk in encoding
+        existing_chunks = encoding[name].get("chunks", None)
+        if len(val.shape) > 0 and existing_chunks is not None:
             chunks = _get_auto_chunk(val)
-            val_encoding.update({"chunks": chunks})
-        encoding[name] = val_encoding
+            encoding[name]["chunks"] = chunks
+
+        if PREFERRED_CHUNKS in encoding[name]:
+            # Remove 'preferred_chunks', use chunks only instead
+            encoding[name].pop(PREFERRED_CHUNKS)
```

Comment:
Is this block due to the current `zarr_combine` implementation to be dropped later? I couldn't figure out what this is but a full codebase search showed `preferred_chunks` under `zarr_combine.py`

Reply:
Seems like this encoding comes from xarray directly, I was thinking this was problematic, but then I just tried it and seem like it doesn't affect anything, so I'll remove this check. See: https://github.com/pydata/xarray/blob/960f15ce35826b85745adf5a7b5d8f7decbfa16d/xarray/backends/zarr.py#L493